### PR TITLE
Refs Task#387291 refactor, search webhook by name

### PR DESCRIPTION
**Pull request by quanghung97**
## Related Tickets

- [#387291](https://dev.sun-asterisk.com/issues/387291)

## What's this PR do ?

- [x] Search Webhook by name
- [x] Remove datatable in list admin webhooks
## Performance

- [x] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [ ] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [ ] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

File: config/paginate.php [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?php
+return [
+    'perPage' => 15,
```
Comment (ghost) [inline config/paginate.php:3]: đổi thành webhooksPerPage đi cậu vì lỡ sau này mình có record khác chỉ hiển thị 10 records thì không tái sử dụng cái này được

Comment (quanghung97) [reply config/paginate.php:3]: quy ước chung là con số 15, còn 1 số trang đặc biệt con số là 10 thì các page khác phải đặt biến như là abcPerPage để dùng chứ nhỉ

**Review by tamtt-1031 — APPROVED** [review]

File: app/Http/Controllers/Admin/WebhookController.php [hunk 2]
```diff
@@ -14,9 +15,15 @@ public function __construct(WebhookRepository $webhookRepository)
         $this->webhookRepository = $webhookRepository;
     }
 
-    public function index()
+    public function index(Request $request)
     {
-        $webhooks = $this->webhookRepository->getAll();
+        $keyword = $request->get('search');
+        $perPage = config('paginate.perPage');
+        if (!empty($keyword)) {
```
Comment (ghost) [inline app/Http/Controllers/Admin/WebhookController.php]: cần đoạn if-else này không, thấy ở dưới [này](https://github.com/sun-asterisk-research/chatwork-forwarder/pull/39/files#diff-e92e5658970f83bbb80614cb9918094cR29) check rồi mà

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Review by nguyenthinhhn — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**